spec/rental_spec.rb

```diff
@@ -0,0 +1,28 @@
+require_relative '../rentals'
+require_relative '../student'
+require_relative '../book'
+
+describe Rental do
+  context 'Testing the Rental class' do
+    before(:each) do
+      @person = Student.new(nil, 23, 'Sam')
+      @book = Book.new('Legend', 'John')
+      @rental = Rental.new('2023', @person, @book)
+    end
+
+    it 'should add a new rental' do
+      name = @rental.person.name
+      title = @rental.book.title
+      date = @rental.date
+      expect(name).to eq('Sam')
+      expect(title).to eq('Legend')
+      expect(date).to eq('2023')
+    end
+
+    it 'should show that "rentals" in Book and Person are updated' do
+      expect(@person.rentals.length.positive?).to eq(true)
+      expect(@book.rentals.length.positive?).to eq(true)
+      expect(@person.rentals[0].person.name).to eq('Sam')
+    end
+  end
```

Comment:
- [ ]  @emmanuelsteven Great work adding tests for the rental class, kindly consider adding tests for the to_hash  method inside the student class.

Reply:
- [ ]  @emmanuelsteven Great work adding tests for your classes, however you for tot add tests for that you created while implementing the Decorator design pattern, this is part of the project requirements as shown below
![image](https://github.com/emmanuelsteven/OOP-school-library/assets/61408860/2e5029ec-ffc2-46a7-a761-85ac1a7f3638)